src/helpers/pairs.ts

```diff
@@ -0,0 +1,104 @@
+import { remapf, remapColor, rgba as RGBA } from "../interpolate";
+import { normalizeColor, rgba } from "./normalizeColor";
+
+export interface CSSPairs {
+  from: React.CSSProperties;
+  to: React.CSSProperties;
+}
+
+export type CSSProperty = keyof React.CSSProperties;
+
+const parsePair = (pairs: React.CSSProperties) =>
+  Object.entries(pairs).map(([k, v]) => ({
+    property: k as CSSProperty,
+    value: v
+  }))[0];
+
+export const parsePairs = ({ from, to }: CSSPairs) => ({
+  from: parsePair(from),
+  to: parsePair(to)
+});
+
+type Interpolator<T> = (params: {
+  readonly range: [number, number];
+  readonly domain: [T, T];
+  readonly value: number;
+}) => T;
```

Comment:
All interpolator functions _must_ match this shape. Having a normalized shape for interpolators like this helps to keep our functions much cleaner.